nitransforms/linear.py

```diff
@@ -402,119 +402,6 @@ def to_filename(self, filename, fmt="X5", moving=None):
-
-        Parameters
-        ----------
-        spatialimage : `spatialimage`
-            The image object containing the data to be resampled in reference
-            space
-        reference : spatial object, optional
-            The image, surface, or combination thereof containing the coordinates
-            of samples that will be sampled.
-        order : int, optional
-            The order of the spline interpolation, default is 3.
-            The order has to be in the range 0-5.
-        mode : {"constant", "reflect", "nearest", "mirror", "wrap"}, optional
-            Determines how the input image is extended when the resamplings overflows
-            a border. Default is "constant".
-        cval : float, optional
-            Constant value for ``mode="constant"``. Default is 0.0.
-        prefilter: bool, optional
-            Determines if the image's data array is prefiltered with
-            a spline filter before interpolation. The default is ``True``,
-            which will create a temporary *float64* array of filtered values
-            if *order > 1*. If setting this to ``False``, the output will be
-            slightly blurred if *order > 1*, unless the input is prefiltered,
-            i.e. it is the result of calling the spline filter on the original
-            input.
-
-        Returns
-        -------
-        resampled : `spatialimage` or ndarray
-            The data imaged after resampling to reference space.
-
-        """
-
-        if reference is not None and isinstance(reference, (str, Path)):
-            reference = _nbload(str(reference))
-
-        _ref = (
-            self.reference if reference is None else SpatialReference.factory(reference)
-        )
-
-        if isinstance(spatialimage, (str, Path)):
-            spatialimage = _nbload(str(spatialimage))
-
-        # Avoid opening the data array just yet
-        input_dtype = get_obj_dtype(spatialimage.dataobj)
-        output_dtype = output_dtype or input_dtype
-
-        # Prepare physical coordinates of input (grid, points)
-        xcoords = _ref.ndcoords.astype("f4").T
-
-        # Invert target's (moving) affine once
-        ras2vox = ~Affine(spatialimage.affine)
-
-        if spatialimage.ndim == 4 and (len(self) != spatialimage.shape[-1]):
-            raise ValueError(
-                "Attempting to apply %d transforms on a file with "
-                "%d timepoints" % (len(self), spatialimage.shape[-1])
-            )
-
-        # Order F ensures individual volumes are contiguous in memory
-        # Also matches NIfTI, making final save more efficient
-        resampled = np.zeros(
-            (xcoords.shape[0], len(self)), dtype=output_dtype, order="F"
-        )
-
-        dataobj = (
-            np.asanyarray(spatialimage.dataobj, dtype=input_dtype)
-            if spatialimage.ndim in (2, 3)
-            else None
-        )
-
-        for t, xfm_t in enumerate(self):
```

Comment:
I've been holding on this one - the new implementation is 4D, while the old implementation was 3D+t (meaning that you could not resample off-gried on the fourth dimension).

We should make sure we meet today @effigies, @jmarabotto and discuss this.

cc/ @sgiavasis who may be interested in learning more.